File: war3map/Generic/Logger.ts
export class Logger {
    static doLogVerbose = false;
    static doLogDebug = true;
    static doLogWarning = true;
    static doLogCritical = true;

    public static LogVerbose(...params: any[]) {
        if (this.doLogVerbose) {
            print("Verbose: ", ...params);
        }
    }

    public static LogDebug(...params: any[]) {
        if (this.doLogDebug) {
            print("Debug: ", ...params);
        }
    }

    public static LogWarning(...params: any[]) {
        if (this.doLogWarning) {
            print("Warning: ", ...params);
        }
    }

    public static LogCritical(...params: any[]) {
        if (this.doLogCritical) {
            print("Critical: ", ...params);
        }
    }
}

File: src/war3map/Generic/Logger.lua
local ____exports = {}
____exports.Logger = {}
local Logger = ____exports.Logger
Logger.name = "Logger"
Logger.__index = Logger
Logger.prototype = {}
Logger.prototype.__index = Logger.prototype
Logger.prototype.constructor = Logger
function Logger.new(...)
    local self = setmetatable({}, Logger.prototype)
    self:____constructor(...)
    return self
end
function Logger.prototype.____constructor(self)
end
function Logger.LogVerbose(self, ...)
    local params = ({...})
    if self.doLogVerbose then
        print("Verbose: ", table.unpack(params))
    end
end
function Logger.LogDebug(self, ...)
    local params = ({...})
    if self.doLogDebug then
        print("Debug: ", table.unpack(params))
    end
end
function Logger.LogWarning(self, ...)
    local params = ({...})
    if self.doLogWarning then
        print("Warning: ", table.unpack(params))
    end
end
function Logger.LogCritical(self, ...)
    local params = ({...})
    if self.doLogCritical then
        print("Critical: ", table.unpack(params))
    end
end
Logger.doLogVerbose = false
Logger.doLogDebug = true
Logger.doLogWarning = true
Logger.doLogCritical = true
return ____exports
